slip.js

```diff
@@ -454,21 +454,25 @@ window['Slip'] = (function(){
 
                     onEnd: function() {
                         var move = this.getTotalMovement();
+                        var i, spliceIndex;
                         if (move.y < 0) {
-                            for(var i=0; i < otherNodes.length; i++) {
+                            for (i=0; i < otherNodes.length; i++) {
                                 if (otherNodes[i].pos > move.y) {
-                                    this.dispatch(this.target.node, 'reorder', {spliceIndex:i, insertBefore:otherNodes[i].node, originalIndex: originalIndex});
                                     break;
                                 }
                             }
+                            spliceIndex = i;
                         } else {
-                            for(var i=otherNodes.length-1; i >= 0; i--) {
+                            for (i=otherNodes.length-1; i >= 0; i--) {
```

Comment:
Think of these loops as simply finding the first index that meets the condition.  And if the condition is never met, the index that falls out of the loop is the one to use.